### PR TITLE
Add support for configuring PropertyNamingStrategy on the Context ObjectMapper

### PR DESCRIPTION
**Pull request by msanssouci**
The default property naming strategy is lower camel case and I need to change it to snake case. Being able to configure a different Overriding the ObjectMapper in the application context seems too intrusive for me. Configuring the PropertyNamingStrategy of the Context ObjectMapper in the application configuration makes the most sense to me.

Would it make sense to add this to [JacksonFeatures](https://github.com/micronaut-projects/micronaut-core/blob/master/runtime/src/main/java/io/micronaut/jackson/annotation/JacksonFeatures.java) annotation as well?

### CODE REVIEW

File: runtime/src/main/java/io/micronaut/jackson/JacksonPropertyNamingStrategy.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.jackson;
+
+import com.fasterxml.jackson.databind.PropertyNamingStrategy;
+
+public enum JacksonPropertyNamingStrategy {
```
Comment (jameskleeh) [inline runtime/src/main/java/io/micronaut/jackson/JacksonPropertyNamingStrategy.java]: Why does this class need to exist? You can just use the jackson type in the config class

File: runtime/src/main/java/io/micronaut/jackson/JacksonConfiguration.java [hunk 2]
```diff
@@ -218,4 +227,15 @@ public void setSerializationInclusion(JsonInclude.Include serializationInclusion
             this.serializationInclusion = serializationInclusion;
         }
     }
+
+    /**
+     * Sets the property naming strategy.
+     *
+     * @return
+     */
+    public void setPropertyNamingStrategy(String propertyNamingStrategy) {
```
Comment (jameskleeh) [inline runtime/src/main/java/io/micronaut/jackson/JacksonConfiguration.java]: The type here should be the jackson type. You can write a converter to convert the string to the type.

File: runtime/src/main/java/io/micronaut/jackson/JacksonConfiguration.java [hunk 3]
```diff
@@ -126,6 +128,13 @@ public int getArraySizeThreshold() {
         return arraySizeThreshold;
     }
 
+    /**
+     * @return The property naming strategy
+     */
+    public JacksonPropertyNamingStrategy getPropertyNamingStrategy() {
```
Comment (jameskleeh) [inline runtime/src/main/java/io/micronaut/jackson/JacksonConfiguration.java]: The return type should be the jackson type

**Comment by jameskleeh** [comment]
This should not be added to the features annotation because it isnt a jackson feature

**Comment by jameskleeh** [comment]
The new method should not default to anything. Only set the method if its set in configuration. This ensures the default value that jackson uses will be used.

File: runtime/src/main/java/io/micronaut/jackson/JacksonConfiguration.java (hunk 2 above)
Comment (msanssouci) [reply runtime/src/main/java/io/micronaut/jackson/JacksonConfiguration.java]: I will add a converter.

Comment (msanssouci) [reply runtime/src/main/java/io/micronaut/jackson/JacksonConfiguration.java]: @jameskleeh Are you referring to creating an implementation of a io.micronaut.core.convert.TypeConverter?

Comment (jameskleeh) [reply runtime/src/main/java/io/micronaut/jackson/JacksonConfiguration.java]: @msanssouci Yep!

**Pull request merged**